article/_posts/2024-05-31-marl-seac.md

```diff
@@ -0,0 +1,97 @@
+	- Assume: $$A = A^1 = ... = A^N$$ and $$O = O^1 = ... = O^N$$ 
+- **PG and Actor-Critic**
+	- Classic policy loss: $$L(\phi_i) = -\log \pi(a^i_t \mid o^i_t; \phi_i) (r^i_t + \gamma V(o^i_{t+1}; \theta_i) - V(o^i_t; \theta_i))$$
+	- Classic value loss: $$L(\theta_i) = \| V(o^i_t; \theta_i) - y_i \|^2  \text{  with  } y_i = r_t^i + \gamma V(o_{t+1}^i; \theta_i)$$
+
+# Shared Experience Actor-Critic
+
+## Method
+
+- Assumption: local policy gradients provide useful learning directions for all agents. Gradients come from the policy, so the reward isn't involved => agents can have different rewards.
+- Need to add importance sampling since updates can come from other policies. **Using the following, each agent is trained on both on-policy data while also using the off-policy data collected by all other agents at each training step.**
+	- New Policy loss that accounts for updates from other agents (k): $$L(\phi_i) = -\log \pi(a^i_t \mid o^i_t; \phi_i) (r^i_t + \gamma V(o^i_{t+1}; \theta_i) - V(o^i_t; \theta_i)) - \lambda \sum_{k \ne i} \frac{\pi(a^k_t \mid o^k_t;\phi_i)}{\pi(a^k_t \mid o^k_t;\phi_k)} \log \pi(a^k_t \mid o^k_t; \phi_i) (r^k_t + \gamma V(o^k_{t+1}; \theta_i) - V(o^k_t; \theta_i))$$
+	- New Value loss that accounts for updates from other agents (k): $$L(\theta_i) = \| V(o^i_t; \theta_i) - y_i \|^2 + \lambda \sum_{k \ne i}\frac{\pi(a_t^k \mid o_t^k;\phi_i)}{\pi(a_t^k \mid o_t^k; \phi_k)} \| V(o^i_t; \theta_i) - y_k^i \|^2 \text{  with  } y_i^k = r_t^k + \gamma V(o_{t+1}^k; \theta_i)$$
+	- $$\lambda$$: weight of experience from other agents (doesn't really matter, use $$\lambda = 1$$)
+	
+	![SEAC Pseudo-code](/article/images/marl-seac/SEAC-Algo.png) 
+
+## Experiments
+
+- Environments
+	- Predator Prey: If at least 2 predators are close to the prey they receive a reward of 1, if they leave the map they receive negative reward, otherwise the reward is zero.
+	- SMAC (SC2 Multi-Agent Challenge): variant that the agent controls several marines with a reward of 1 upon winning and a reward of -1 if facing defeat.
+	- Label-Based Foraging: rewards from fraction of items collected per episode compared to other agents.
+	- Multi-Robot Warehouse: POMDP (limited view), very sparse reward signal only seen upon making a successful delivery.
+- Baseline algorithms
+	- Independent Actor-Critic (IAC): independent policies with independent learning.
+	- Shared Network Actor-Critic (SNAC): single policy that is updated online by each agent, each agent executes a copy of that policy.
+- Algorithm details
+	- 5-step returns with 4 environments were sampled in batches to the optimiser.
+	- Entropy regularization term (relative to agent $$i$$) added to each agent $$i$$ loss
+
+## Results
+
+- PP only SEAC learns successfully.
+- SMAC means are close to 0.0. The agents learned to ran away from the enemy instead of winning the battle. SOTA methods designed for this environment don't even solve this sparsely rewarded task.
```

Comment:
```suggestion
- SMAC means are close to 0.0. The agents learned to run away from the enemy instead of winning the battle. SOTA methods designed for this environment don't even solve this sparsely rewarded task.
```